diff --git a/third_party/libpng/BUILD.gn b/third_party/libpng/BUILD.gn
index 81648e3b2a810..4252f0d8acb7a 100644
--- a/third_party/libpng/BUILD.gn
+++ b/third_party/libpng/BUILD.gn
@@ -91,6 +91,10 @@ source_set("libpng_sources") {
       "mips/filter_msa_intrinsics.c",
       "mips/mips_init.c",
     ]
+  } else if (current_cpu == "ppc64") {
+    cflags += [
+      "-DPNG_POWERPC_VSX_OPT=0",
+    ]
   }

   configs -= [ "//build/config/compiler:chromium_code" ]

