diff -up chromium-133.0.6943.53/third_party/rust/crc32fast/v1/BUILD.gn.me chromium-133.0.6943.53/third_party/rust/crc32fast/v1/BUILD.gn
--- chromium-133.0.6943.53/third_party/rust/crc32fast/v1/BUILD.gn.me	2025-02-11 00:13:06.132152874 +0100
+++ chromium-133.0.6943.53/third_party/rust/crc32fast/v1/BUILD.gn	2025-02-11 00:28:22.785233997 +0100
@@ -41,7 +41,6 @@ cargo_crate("lib") {
   proc_macro_configs += [ "//build/config/compiler:no_chromium_code" ]
   deps = [ "//third_party/rust/cfg_if/v1:lib" ]
   features = [
-    "nightly",
     "std",
   ]
   rustflags = [
