diff -up chromium-132.0.6834.57/pdf/pdfium/pdfium_api_wrappers.cc.me chromium-132.0.6834.57/pdf/pdfium/pdfium_api_wrappers.cc
--- chromium-132.0.6834.57/pdf/pdfium/pdfium_api_wrappers.cc.me	2025-01-10 11:17:44.598459092 +0100
+++ chromium-132.0.6834.57/pdf/pdfium/pdfium_api_wrappers.cc	2025-01-10 11:18:55.910222909 +0100
@@ -147,7 +147,7 @@ std::u16string GetPageObjectMarkName(FPD
 
   // Number of characters, including the NUL.
   const size_t expected_size = base::checked_cast<size_t>(buflen_bytes / 2);
-  PDFiumAPIStringBufferAdapter adapter(&name, expected_size,
+  PDFiumAPIStringBufferAdapter<std::u16string> adapter(&name, expected_size,
                                        /*check_expected_size=*/true);
   unsigned long actual_buflen_bytes = 0;  // NOLINT(runtime/int)
   bool result =
