diff -up chromium-133.0.6943.53/chrome/browser/compose/compose_session.cc.me chromium-133.0.6943.53/chrome/browser/compose/compose_session.cc
--- chromium-133.0.6943.53/chrome/browser/compose/compose_session.cc.me	2025-02-10 14:16:34.810693753 +0100
+++ chromium-133.0.6943.53/chrome/browser/compose/compose_session.cc	2025-02-10 14:16:39.747791800 +0100
@@ -543,7 +543,7 @@ void ComposeSession::RequestWithSession(
   // execution in case request fails.
   compose::LogComposeRequestReason(request_reason);
 
-  optimization_guide::ModelExecutionSessionCallbackWithLogging callback =
+  optimization_guide::ModelExecutionSessionCallbackWithLogging<optimization_guide::proto::ComposeLoggingData> callback =
       base::BindRepeating(&ComposeSession::ModelExecutionCallback,
                           weak_ptr_factory_.GetWeakPtr(),
                           std::move(request_timer), request_id_, request_reason,
